frontend/templates/Chat/DiscoveryLibraryUI/DiscoveryLibraryUI.jsx

```diff
@@ -0,0 +1,144 @@
+import React, { useEffect, useState } from 'react';
+
+import {
+  Card,
+  CardActionArea,
+  CardContent,
+  Grid,
+  IconButton,
+  Typography,
+} from '@mui/material';
+
+import { useDispatch, useSelector } from 'react-redux';
+
+import DiscoveryIcon from '@/assets/svg/add-block2.svg';
+
+import imageCover1 from '@/assets/svg/ImageCover1.svg';
+import imageCover2 from '@/assets/svg/ImageCover2.svg';
+import imageCover3 from '@/assets/svg/ImageCover3.svg';
+import imageCover4 from '@/assets/svg/ImageCover4.svg';
+import AvatarImage from '@/assets/svg/ReadyPlayerMeAvatar.svg';
+import StarGroupIcon from '@/assets/svg/starGroupIcon.svg';
+import UnionIcon from '@/assets/svg/Union.svg';
+
+import styles from './styles';
+
+import { resetChat, setInput } from '@/redux/slices/chatSlice';
+
+const DiscoveryLibraryUI = (props) => {
+  const { show, handleSendMessage, selectedPrompt } = props;
+  const { data: user } = useSelector((state) => state.user);
+  const [customPrompts, setCustomPrompts] = useState([]);
+  const dispatch = useDispatch();
+
+  const getRandomImage = () => {
+    const imageUrls = [
+      imageCover1,
+      imageCover2,
+      imageCover3,
+      imageCover4,
+      // Add more image URLs as needed
+    ];
+
+    const RandomImage = imageUrls[Math.floor(Math.random() * imageUrls.length)];
+    return <RandomImage {...styles.backImageProps} />;
+  };
+
+  useEffect(() => {
+    // Define your custom prompts here
+    const categorizePrompts = () => [
```

Comment:
Given that these a default prompts, it should be moved either to the top of this file as a constant and not a function or move it to a file in the constants folder